tests/DatabaseTest.php

```diff
@@ -20,6 +20,30 @@
 
 	use TestCaseTrait;
 
+	/**
+	 * Creates .htconfig.php for bin/worker.php execution
+	 */
+	protected function setUp()
+	{
+		parent::setUp();
+
+		$base_config_file_name = 'htconfig.php';
+		$config_file_name = '.htconfig.php';
+
+		$base_config_file_path = stream_resolve_include_path($base_config_file_name);
+		$config_file_path = dirname($base_config_file_path) . DIRECTORY_SEPARATOR . $config_file_name;
+
+		$config_string = file_get_contents($base_config_file_path);
+
+		$config_string = str_replace('die(', '// die(', $config_string);
+		$config_string = str_replace('your.mysqlhost.com', 'localhost', $config_string);
+		$config_string = str_replace('mysqlusername'     , getenv('USER'), $config_string);
+		$config_string = str_replace('mysqlpassword'     , getenv('PASS'), $config_string);
+		$config_string = str_replace('mysqldatabasename' , getenv('DB'), $config_string);
+
+		file_put_contents($config_file_path, $config_string);
```

Comment:
This means that when I run the tests on my local setup, it replaces my configuration file without a warning!

Wouldn't it be possible to use a different name for the test config file?
(Or have a way to construct the config object from the environment variables without a config file, which would be way cleaner IMHO.)

Also, do we need both this and the `getConnection()` method which also handles the environment variables?

Reply:
The problem is that some tests are triggering an execution of `bin/worker.php` and this file is currently looking for the database configuration in `.htconfig.php` explicitly. If it's missing or if the database config variables are missing, it's outputting a notice and the tests are failing.

The configuration file in itself is supposed to go with #5295, but the PR over there is so big that I've been wanting to break it down in reviewable chunks.

I see we're using `MYSQL_*` environment variables to override local settings, the tests probably should use them instead.

Reply:
Oh, I didn't see the `MYSQL_*` variables, it is indeed a good idea to use them in tests as well!

I don't use Windows anymore but I remember a `set` command that can be used to set environment variables for the current session.

As a workaround, you could also manually add `putenv()` calls in `tests/bootstrap.php`.